charts/nextjs/templates/deployment.yaml

```diff
@@ -32,22 +32,54 @@ spec:
       securityContext:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+      {{- if .Values.postgresql.enabled }}
+      initContainers:
+        - name: migrate
+          {{- if .Values.image.repository }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          {{- else }}
+          image: "{{ .Values.image.repository | default "your-nextjs-app" }}:{{ .Values.image.tag | default "latest" }}"
+          {{- end }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command: ["npm", "run", "db:migrate"]
+          env:
+            - name: DATABASE_URL
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.postgresql.auth.existingSecret | default (printf "%s-database" (include "nextjs.fullname" .)) }}
```

Comment:
This should always be use the existingSecret, no default, infact it shoudl throw an error if the exsting secret is not defined